src/components/visualization/mermaidViz/mermaidViz.tsx

```diff
@@ -0,0 +1,69 @@
+import './mermaidViz.css'
+
+import Stack from '@mui/material/Stack'
+import type { Theme } from '@mui/material/styles'
+import Typography from '@mui/material/Typography'
+import useTheme from '@mui/system/useTheme'
+import mermaid from 'mermaid'
+import React, { useEffect, useRef, useState } from 'react'
+
+import type { MermaidData } from '../../types'
+
+/** The `MermaidViz` component leverages [Mermaid.js](https://mermaid.js.org/) to create dynamic and interactive diagrams, including flowcharts, sequence diagrams, class diagrams, and more. It is ideal for visualizing complex data and processes in a clear and structured manner. */
+function MermaidViz(props: MermaidData) {
+  const mermaidRef = useRef<HTMLDivElement>(null)
+  const [errorMessage, setErrorMessage] = useState<string>()
+  const [isProcessed, setIsProcessed] = useState<boolean>(false)
+  const rusticTheme: Theme = useTheme()
+  const mermaidTheme = rusticTheme.palette.mode === 'dark' ? 'dark' : 'neutral'
+  const defaultFont = rusticTheme.typography.fontFamily
+
+  useEffect(() => {
+    if (mermaidRef.current) {
+      mermaid.initialize({
+        theme: mermaidTheme,
+        themeVariables: { fontFamily: defaultFont },
+        ...props.config,
+      })
+
+      mermaid
+        .run({
+          querySelector: '.rustic-mermaid',
```

Comment:
This query selector is too generic and will apply to all the elements on the page with this class.